### PR TITLE
Add flashinfer.fused_rmsnorm_silu() using CuDNN OSS 

### PR DESCRIPTION
**Pull request by kahyunnam**
## 📌 Description

**Do not merge until cudnn frontend 1.20 released**

Adds `flashinfer.fused_rmsnorm_silu()` — a single-kernel fusion of RMS normalization + SiLU activation, using cuDNN frontend OSS `Sm100RmsNormSiluEngine` as backend. The kernel is JIT-compiled via NVRTC at runtime and dispatched through the cuDNN graph API (`heur_mode.OPENSOURCE`), following the same integration pattern as `flashinfer/cudnn/prefill.py` and `decode.py`.

**Tuned and optimized for the WAN VAE decoder problem sizes on B200 (SM100).** Functional on SM80+ GPUs with a conservative fallback heuristic for non-tuned problem sizes.

## 🔍 Related Issues

[https://github.com/flashinfer-ai/flashinfer/issues/2571](https://github.com/flashinfer-ai/flashinfer/issues/2571 )

## 🚀 Pull Request Checklist

Thank you for contributing to FlashInfer! Before we review your pull request, please make sure the following items are complete.

### ✅ Pre-commit Checks

- [x] I have installed `pre-commit` by running `pip install pre-commit` (or used your preferred method).
- [x] I have installed the hooks with `pre-commit install`.
- [x] I have run the hooks manually with `pre-commit run --all-files` and fixed any reported issues.

> If you are unsure about how to set up `pre-commit`, see [the pre-commit documentation](https://pre-commit.com/).

## 🧪 Tests

- [x] Tests have been added or updated as needed.
- [ ] All tests are passing (`unittest`, etc.).

## Summary by CodeRabbit

* **New Features**
  * Added a fused RMSNorm + SiLU operation with bf16, FP8 and NVFP4 output support and cuDNN-backed acceleration where available
  * Re-exported the new operation in the package API and added a benchmarking script to measure performance across tensor sizes

* **Tests**
  * Added comprehensive tests covering bf16/FP8/NVFP4 paths, preallocated outputs, and varied shapes/configurations

* **Chores**
  * Updated cuDNN frontend version constraint in requirements.txt

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -3,7 +3,7 @@ click
 einops
 ninja
 numpy
-nvidia-cudnn-frontend>=1.13.0
+nvidia-cudnn-frontend>=1.13.0 # would need to update to 1.19.0 for rmsnorm+silu once it's released, or add a version check in the module.
```
Comment (kahyunnam) [inline requirements.txt]: (should be 1.20)

File: flashinfer/cudnn/norm.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        graph.build([cudnn.heur_mode.OPENSOURCE])
+
+        workspace_size = graph.get_workspace_size()
+
+    return graph, X, scale, epsilon, Z, workspace_size
+
+
+def cudnn_fused_rmsnorm_silu(
+    input: torch.Tensor,
+    weight: torch.Tensor,
```
Comment (Anerudhan) [inline flashinfer/cudnn/norm.py]: Lets call this input_scale or scale and not weight.

Should have an equation for the underlying operation?

File: flashinfer/cudnn/norm.py [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+import functools
+from typing import Optional
+import torch
+import cudnn
+
+_TORCH_TO_CUDNN_OUTPUT_DTYPE = {
+    torch.bfloat16: cudnn.data_type.BFLOAT16,
+    torch.float8_e4m3fn: cudnn.data_type.FP8_E4M3,
+    torch.uint8: cudnn.data_type.FP4_E2M1,  # packed FP4: 2 elements per byte
```
Comment (Anerudhan) [inline flashinfer/cudnn/norm.py]: Should we mention torch.nvfp4?
https://github.com/pytorch/ao/blob/main/torchao/prototype/qat/nvfp4.py

File: flashinfer/cudnn/norm.py [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+}
+
+
+@functools.lru_cache(maxsize=128)
+def _build_rmsnorm_silu_graph(
```
Comment (Anerudhan) [inline flashinfer/cudnn/norm.py:34]: Given that the kernel itself is independent of the number of tokens, should the cache be independent of num_tokens?

File: tests/norm/test_fused_rmsnorm_silu.py [hunk 5]
```diff
@@ -0,0 +1,410 @@
...
+    if _get_sm() < 100:
+        pytest.skip("NVFP4 output requires SM100+ (Blackwell)")
+
+    torch.manual_seed(42)
+    x = torch.randn(num_tokens, C, dtype=torch.bfloat16, device="cuda") * 5.0 + 5.0
+    w = torch.rand(C, dtype=torch.bfloat16, device="cuda") * 1.5 + 0.5
+    eps = 1e-6
+
+    # Build cuDNN graph directly so we control the workspace
+    graph = cudnn.pygraph(
```
Comment (Anerudhan) [inline tests/norm/test_fused_rmsnorm_silu.py]: Shouldnt you be testing `fused_rmsnorm_silu` here?

File: flashinfer/cudnn/norm.py (hunk 2 above)
Comment (kahyunnam) [reply flashinfer/cudnn/norm.py]: https://github.com/flashinfer-ai/flashinfer/pull/2691/changes/6696d09a00a694d8d7595c4d72cce5f92319d5bc

File: flashinfer/cudnn/norm.py (hunk 3 above)
Comment (kahyunnam) [reply flashinfer/cudnn/norm.py]: https://github.com/flashinfer-ai/flashinfer/pull/2691/commits/c3bf930745e566acb938b9b20825fcda7d63fce6

File: flashinfer/cudnn/norm.py (hunk 4 above)
Comment (kahyunnam) [reply flashinfer/cudnn/norm.py:34]: https://github.com/flashinfer-ai/flashinfer/pull/2691/commits/3d915f7c13cabce6a293b775a0c92f849c97ee90

File: tests/norm/test_fused_rmsnorm_silu.py (hunk 5 above)
Comment (kahyunnam) [reply tests/norm/test_fused_rmsnorm_silu.py]: https://github.com/flashinfer-ai/flashinfer/pull/2691/commits/d54dc86a97e57ce78951bfa739bf87b48a38d67c
